### PR TITLE
A spyOn implementation for whiskey. (in progress)

### PR DESCRIPTION
**Pull request by btipling**

### CODE REVIEW

File: lib/common.js [hunk 1]
```diff
@@ -544,3 +545,73 @@ exports.TestFile = TestFile;
...
+/**
+ * @constructor
+ */
+function SpyOn (){
+  /**
+   * This tracks the number of times a function has been fired.
+   * @param {Object.<number>}
+   * @private
+   */
+  this.callCount_ = {};
```
Comment (Kami) [inline lib/common.js]: Minor style issue - we prefix 'private' attributes with _ (_callCount) instead of suffix them.

File: lib/common.js [hunk 2]
```diff
@@ -544,3 +545,73 @@ exports.TestFile = TestFile;
...
+  this.funcMap_[funcName] = func;
+  return this;
+};
+
+/**
+ * @param {string} funcName The key to clear.
+ * @param {Object} context The context in which the function should execute.
+ * @param {Function} opt_func The optional function to reapply to the context.
+ */
+SpyOn.prototype.clear = function (funcName, context, opt_func) {
```
Comment (Kami) [inline lib/common.js]: Another minor style issue: `opt_func` -> `optFunc`.

File: lib/common.js (hunk 1 above)
Comment (btipling) [inline lib/common.js]: Fixed.

File: lib/common.js (hunk 2 above)
Comment (btipling) [inline lib/common.js]: Fixed.

**Pull request merged**